platform/linux-generic/arch/aarch64/odp_random.h

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2021, ARM Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_AARCH64_RANDOM_H_
+#define ODP_AARCH64_RANDOM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/spec/random.h>
+#include <odp/autoheader_internal.h>
+
+#include <stdint.h>
+
+odp_random_kind_t _odp_random_max_kind_generic(void);
+int32_t _odp_random_true_data_generic(uint8_t *buf, uint32_t len);
+int32_t _odp_random_crypto_data_generic(uint8_t *buf, uint32_t len);
+
+#ifdef __ARM_FEATURE_RNG
+
+#if __ARM_FEATURE_UNALIGNED != 1
+#error Unaligned access not supported
+#endif
+
+static inline int _odp_random_max_kind(void)
+{
+	return ODP_RANDOM_TRUE;
+}
+
+static inline int _odp_rndr(unsigned long *v)
```

Comment:
Type of the pointer should be uint64_t. The instruction outputs 64bits, but in armv8 long type may be either 64 or 32 bits depending if LP64 or LLP64 model is used. Maybe long is in practice always 64bits, but usage of it adds confusion about output data size/align. Remember to change also (unsigned long *) casts to uint64_t cast under.

_odp_rndrrs() prototype needs the same change.

Reply:
This point was something even me and Govind had discussed and we had concluded after some research that `unsigned long` will most likely end up being 64-bits. But, to avoid any ambiguity, I will make the change to use `uint64_t` instead.